src/dmd/initsem.d

```diff
@@ -363,10 +363,10 @@ extern(C++) Initializer initializerSemantic(Initializer init, Scope* sc, ref Typ
             if (length > i.dim)
                 i.dim = length;
         }
-        if (t.ty == Tsarray)
+        if (auto tsa = t.isTypeSArray())
         {
-            uinteger_t edim = (cast(TypeSArray)t).dim.toInteger();
-            if (i.dim > edim)
+            uinteger_t edim = tsa.dim.toInteger();
+            if (i.dim > edim && !(tsa.isIncomplete() && (sc.flags & SCOPE.Cfile)))
```

Comment:
```suggestion
            if (i.dim > edim && !((sc.flags & SCOPE.Cfile) && tsa.isIncomplete()))
```

Reply:
Rationale?

Reply:
This code will be executed for both D and C code, however, the former should be the common case, in which case `tsa.isIncomplete` will be evaluated only if necessary.

Reply:
`isComplete` is a trivial function.